tests/ValueFormatters/ValueFormatterTestBase.php

```diff
@@ -56,7 +56,7 @@ public function testValidFormat(
 			$formatter = $this->getInstance( $options );
 		}
 
-		$this->assertEquals( $expected, $formatter->format( $value ) );
+		$this->assertSame( $expected, $formatter->format( $value ) );
```

Comment:
I do not predict that we will ever have a formatter that returns anything that is not a string. I'd ague that a formatter will by definition always return a string. Maybe a string plus auxiliary information in a nested structure? Possible. But then that is the strange exception that needs special handling. Assuming that format() will return a string seems sensible to me.